src/courseware/course/CourseLicense.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCopyright } from '@fortawesome/free-regular-svg-icons';
+import {
+  faCreativeCommons,
+  faCreativeCommonsBy,
+  faCreativeCommonsNc,
+  faCreativeCommonsNd,
+  faCreativeCommonsSa,
+} from '@fortawesome/free-brands-svg-icons';
+
+const CreativeCommonsLicenseTags = {
+  by: {
+    text: 'Attribution',
+    icon: faCreativeCommonsBy,
+  },
+  nc: {
+    text: 'Noncommercial',
+    icon: faCreativeCommonsNc,
+  },
+  nd: {
+    text: 'No Derivatives',
+    icon: faCreativeCommonsNd,
+  },
+  sa: {
+    text: 'Share Alike',
+    icon: faCreativeCommonsSa,
+  },
+};
+
+function parseLicense(license) {
+  if (license === undefined) {
+    return [null, {}];
+  }
+
+  const colonIndex = license.indexOf(':');
+  if (colonIndex === -1) {
+    // no options, so the entire thing is the license type
+    return [license, {}];
+  }
+
+  const licenseType = license.slice(0, colonIndex).trim();
+  const optionStr = license.slice(colonIndex + 1).trim();
+  let options = {};
+  let version = '';
+  optionStr.split(' ').forEach(option => {
+    if (option.indexOf('=') !== -1) {
+      const [key, value] = option.split('=');
+      if (key === 'ver') {
+        version = value;
+      } else {
+        options[key.toLowerCase()] = value.toLowerCase();
+      }
+    } else {
+      options[option.toLowerCase()] = true;
+    }
+  });
+
+  if (options.length === 0) {
```

Comment:
Isn't this `undefined` for an `Object`? Shouldn't we use something like this instead?
```suggestion
  if (Object.keys(options).length === 0) {
```

Reply:
Done in next commit